src/components/SUSHIControls.js

```diff
@@ -161,69 +161,17 @@ export default function SUSHIControls(props) {
       jsonOutput = '';
     }
 
-    props.onClick(true, jsonOutput, isObject);
+    props.onClick(true, jsonOutput, isObject, false);
   }
 
   async function handleGoFSHClick() {
-    const exInput = {
-      resourceType: 'Observation',
-      id: 'example',
-      status: 'final',
-      code: {
-        coding: [
-          {
-            code: 'bar',
-            system: 'http://foo.org'
-          }
-        ]
-      },
-      subject: {
-        reference: 'Patient/example'
-      }
-    };
-    const exPatient = {
-      resourceType: 'Patient',
-      id: 'MyPatient',
-      name: [
-        {
-          family: 'Smith',
-          given: ['Jane']
-        }
-      ],
-      maritalStatus: {
-        coding: [
-          {
-            code: 'M',
-            display: 'Married'
-          }
-        ]
-      }
-    };
-    const exIG = {
-      resourceType: 'ImplementationGuide',
-      url: 'http://example.org/tests/ImplementationGuide/bigger.ig',
-      fhirVersion: ['4.0.1'],
-      id: 'bigger.ig',
-      name: 'BiggerImplementationGuideForTesting',
-      status: 'active',
-      version: '0.12',
-      dependsOn: [
-        {
-          uri: 'http://hl7.org/fhir/us/core/ImplementationGuide/hl7.fhir.us.core',
-          packageId: 'hl7.fhir.us.core',
-          version: '3.1.0'
-        },
-        {
-          uri: 'http://hl7.org/fhir/us/mcode/ImplementationGuide/hl7.fhir.us.mcode',
-          packageId: 'hl7.fhir.us.mcode',
-          version: '1.0.0'
-        }
-      ]
-    };
+    props.onGoFSHClick('Loading...', true);
+    const gofshInputStrings = [];
+    props.gofshText.forEach((def) => gofshInputStrings.push(JSON.stringify(def)));
```

Comment:
Is this supposed to work in the case that you just start by pasting JSON on the RHS and try to run GoFSH? I hit a bug in that case, and it looks like it is because `gofshText` is initialized to `''`, so `forEach` does not exist on it. If you run SUSHI first from FSH on the LHS, then things work out because `gofshText` becomes an array, so now you can run GoFSH.

Reply:
Oh good call! I think that should work as part of this, but I definitely hadn't tried it, and that sounds like it makes sense that it breaks. Thanks for finding it!

Reply:
I had too restrictive of a check for not updating the text in the editor. It was assuming you'd only be editing after a SUSHI package had come back and was displayed. e10a50f gets rid of that condition and should fix this bug.